spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java

```diff
@@ -96,22 +85,16 @@ private String uriScheme(URI uri) {
 	}
 
 	/**
-	 * Close the current span and emit the ClientReceivedEvent
+	 * Close the current span and log the client received event
 	 */
 	public void finish() {
 		if (!isTracing()) {
 			return;
 		}
-		publish(new ClientReceivedEvent(this, currentSpan()));
+		currentSpan().logEvent(Span.CLIENT_RECV);
```

Comment:
I think that this is done directly, here. I can see without navigation, that this task results in a "cr"